CONTRIBUTING.md

```diff
@@ -4,45 +4,131 @@ We are very open to community contributions and appreciate anything that improve
+
+### Installation
+
+Now fork and clone the repo. From the terminal, run:
+
+```bash
+git clone https://github.com/<your-gh-username>/haystack.git
+```
+
+or use your favourite Git(Hub) client.
+
+Then move into the cloned folder, create a virtualenv, and perform an **editable install**.
+
+```bash
+# Move into the cloned folder
+cd haystack/
+
+# Create a virtual environment
+python3 -m venv venv
+
+# Activate the environment
+source venv/bin/activate
+
+# Upgrade pip (very important!)
+pip install --upgrade pip
+
+# Install Haystack in editable mode
+pip install -e '.[all]'
+```
+
+This will install all the dependencies you need to work on the codebase, plus testing and formatting dependencies.
+
+Last, install the pre-commit hooks with:
+
+```bash
+pre-commit install --hook-type pre-push
+```
+
+This utility will run some tasks right before all `git push` operations. From now on, your `git push` output for Haystack should look something like this:
+
+```
+> git push
+check python ast.....................................(no files to check)Skipped
+check json...........................................(no files to check)Skipped
+check yaml...............................................................Passed
+check toml...........................................(no files to check)Skipped
+fix end of files.........................................................Passed
+trim trailing whitespace.................................................Passed
+check for merge conflicts................................................Passed
+check that scripts with shebangs are executable..........................Passed
+mixed line ending........................................................Passed
+don't commit to branch...................................................Passed
+pretty format json...................................(no files to check)Skipped
+black................................................(no files to check)Skipped
+Update API documentation (slow)..........................................Passed
+Enumerating objects: 14, done.
+Counting objects: 100% (14/14), done.
+Delta compression using up to 16 threads
+Compressing objects: 100% (12/12), done.
+Writing objects: 100% (12/12), 1.06 KiB | 1.06 MiB/s, done.
+Total 12 (delta 8), reused 0 (delta 0), pack-reused 0
+remote: Resolving deltas: 100% (8/8), completed with 2 local objects.
+To github.com:deepset-ai/haystack.git
+   801be454..6fddb35f  my-branch -> my-branch
+```
+
+Note: pre-commit hooks might fail. If that happens to you and you can't understand why, please do the following:
+- Ask for help by opening an issue or reaching out on our Slack channel. We usually give some feedback within a day for most questions.
+- As the last resort, if you are desperate and everything failed, _uninstall the hook_. `pre-commit uninstall -t pre-commit -t pre-push` will remove all checks and let you push. The CI might fail, but at that point we will be able to help.
```

Comment:
I'd do `git commit --no-verify` instead of uninstalling the hook as the last resort. If the failure is transient the overall developer experience is better.